### PR TITLE
PR for Assignment 5

### PR DESCRIPTION
**Pull request by franci-andreea**
Assignment 5
Things I have implemented:
- Admin dashboard using bootstrap
- admin CRUD operations for products
- using cloudinary for uploading images
- admin handle for orders (changing the status from PENDING to COMPLETED)
- add to cart functionality, meaning that a user can place an order on the app

What I am working on right now:
- implementing API

Things that I have to implement:
- CRUD operations for users and orders

### CODE REVIEW

File: app/controllers/products_controller.rb [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    end
+  end
+
+  def edit
+    @product = Product.find(params[:id])
+  end
+
+  def update
+    product = Product.find(params[:id])
+    if product.update(name: params[:name], price: params[:price].to_f, description: params[:description])
```
Comment (paulaapol) [inline app/controllers/products_controller.rb]: `product.update(product_params)`

File: app/controllers/products_controller.rb [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+      flash[:success] = "Product updated successfully!"
+      redirect_to edit_product_path
```
Comment (paulaapol) [inline app/controllers/products_controller.rb:44]: for better readability, we prefer adding a new line before return and redirect

File: app/helpers/users_helper.rb [hunk 3]
```diff
@@ -4,4 +4,8 @@ def gravatar_for(user, size: 80)
     gravatar_url = "https://secure.gravatar.com/avatar/#{gravatar_id}?=s=#{size}"
     image_tag(gravatar_url, alt: user.name, class: "gravatar")
   end
+
+  def isAdmin?(user)
```
Comment (paulaapol) [inline app/helpers/users_helper.rb]: we use snake case in rails so` is_admin`

File: app/helpers/users_helper.rb [hunk 4]
```diff
@@ -4,4 +4,8 @@ def gravatar_for(user, size: 80)
...
+    return (User.roles[user.role] == 1)
```
Comment (paulaapol) [inline app/helpers/users_helper.rb]: `user.role == 1`

File: app/views/admin/_navbar.html.haml [hunk 5]
```diff
@@ -0,0 +1,18 @@
+%nav.navbar.navbar-expand-lg.navbar-dark.bg-dark
+  = link_to "Hello, #{@current_user.name}!", '/admin', class: ".navbar-brand"
+  %button.navbar-toggler{:type => "button", "data-toggle" => "collapse", "data-target" => "#navbarNav", "aria-controls" => "navbarNav", "aria-expanded" => "false", "aria-label" => "Toggle navigation"}
```
Comment (paulaapol) [inline app/views/admin/_navbar.html.haml:5]: please use new syntax `type: "button"` and add space at the beginning and end of brackets {}

File: app/helpers/users_helper.rb (hunk 4 above)
Comment (franci-andreea) [reply app/helpers/users_helper.rb]: I changed to `return (user.admin?)`

**Pull request merged**